### PR TITLE
Factored out SlackMessageService from DataHubUploaderService.

### PR DESCRIPTION
**Pull request by benwarfield-usds**
What it says on the tin.

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/service/SlackMessageService.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    private DataHubConfig _config;
+    private RestTemplateBuilder _builder;
+
+    public SlackMessageService(DataHubConfig config, RestTemplateBuilder builder) {
+        _config = config;
+        _builder = builder;
+    }
+
+    public void sendSlackChannelMessage(String titleMsg, List<String> markupMsgs, Boolean separateMsgs) {
+        if (!_config.getSlackNotifyWebhookUrl().startsWith("https://hooks.slack.com/")) {
```
Comment (TomNUSDS) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/SlackMessageService.java:36]: The service will only be able to post into a single slack channel. We will _eventually_ need some way to have different channels webhookurls?

**Review by TomNUSDS — APPROVED** [review]
Thanks for moving this into its own, reusable service. 

Minor comments but this refactor LGTM.

File: backend/src/main/java/gov/cdc/usds/simplereport/service/DataHubUploaderService.java [hunk 2]
```diff
@@ -108,51 +107,6 @@ private static Date utcStringToDate(String s) {
...
-            LOG.info("slackMessage Title: '{}'  Body: '{}'", titleMsg, markupMsgsForLog);
-        }
-
-        try {
-            RestTemplate restTemplate = new RestTemplate();
-            HttpHeaders headers = new HttpHeaders();
-            headers.setContentType(MediaType.APPLICATION_JSON);
-
-            // json (escaping)
-            // see https://api.slack.com/messaging/webhooks
```
[the remark this answers is not in the document]
Comment (benwarfield-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/service/DataHubUploaderService.java:128]: Eek! Thanks, I was in the process of moving them from where they were to a more appropriate place, and apparently only got half way.

File: backend/src/main/java/gov/cdc/usds/simplereport/service/SlackMessageService.java (hunk 1 above)
Comment (benwarfield-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/service/SlackMessageService.java:36]: Yeah, I didn't want to get ahead of myself, but I can see eventually having the method be `sendSlackChannelMessage(AlertChannelEnumeration whichChannel, String titleMsg, List<String> body)`, and having a lookup table of valid hooks—either that or switching to using the chat API, which I think lets us pick a channel at runtime? I tried not to get too far ahead of my skis on this one, though. 😄

**Pull request merged**